lib/Basic/LaneBasedExecutionQueue.cpp

```diff
@@ -100,6 +100,12 @@ class LaneBasedExecutionQueue : public ExecutionQueue {
   std::mutex queueCompleteMutex;
   bool queueComplete { false };
 
+  /// Background (lane released) task management
+  std::mutex backgroundTaskMutex;
+  unsigned backgroundTaskMax = 0;
+  unsigned backgroundTaskCount = 0;
```

Comment:
Can we just use atomics for these?

Reply:
Hmm, maybe.  The phrasing around the check site would get a little more awkward if we were to try to keep the hard cap semantics while using a compare_exchange.  Could be a soft cap, though.

Reply:
Updated to the soft cap approach (with comment explaining).